include/dali/core/tensor_shape.h

```diff
@@ -795,9 +795,8 @@ struct TensorListShape<DynamicDimensions>
   TensorListShape &operator=(const TensorListShape &) = default;
   TensorListShape &operator=(TensorListShape &&other) {
     shapes = std::move(other.shapes);
-    nsamples = other.size();
-    ndim = other.sample_dim();
-    other.nsamples = 0;
+    nsamples = std::exchange(other.nsamples, 0);
+    ndim = other.sample_dim();  // Sample dim could be static, and it is valid to not zero it.
```

Comment:
IIUC, previously the `nsamples` wasn't moved, but passed by value, so this changes the behaviour. Is that intentional?

Reply:
It doesn't matter if it's moved, it's an int.

Reply:
it is basically `int nsamples; nsamples = int::int(static_cast<int&&>(other.nsamples)); other.nsamples = 0; ` where both int copy and move constructors are trivial (ie. `memcpy`).